src/map/lua/luautils.cpp

```diff
@@ -4862,4 +4863,21 @@ namespace luautils
 
         customMenuContext.erase(PChar->id);
     }
+
+    uint16 GetItemIDByName(std::string const& name)
+    {
+        uint16      id    = 0;
+        const char* Query = "SELECT itemid FROM item_basic WHERE name LIKE '%s%%' LIMIT 1;";
```

Comment:
I'd rather we just require typing the item name fully and correctly vs partial name matching due to the potential of people later thinking this is a bug when `LIKE` can potentially match an item they were not expecting.

Reply:
To expand upon this as well, REM weapons have multiple stages with the same name.  Would suggest logging an error if the result is >1, or somehow handling this

Reply:
forgot about identical names, that's even more important. See also:
[_All the times SE made mistakes and used same name twice, oops_](https://www.ffxiah.com/forum/topic/55156/localization-errors-typos-english-client)

Reply:
I had considered that.  In this case, the LIMIT 1 in the SQL will only return the first match.  Cases like rema would be where the command would be better served by an ID rather than a name.

I could also allow for a wildcard to be passed by the GM giving the command (i.e. !additem genie_w%) and call a different query within the c++ function if it finds a % in the name field.

In any case, if it finds more than one, it will only return the first item it finds.  If it finds nothing, it return 0 and the command issues an error to the user.

Reply:
I think this might be one of those cases where we need to save future developers from themselves.  Limiting from the SQL query will cause unnecessary pain if someone "thinks" it'll return what they want.

Reply:
Ok, pushed some changes.  It will accept the use of % as a wildcard by the user.  It will now also give a warning if more than one item is found that matches the requested item.  Some example outputs attached.
![image](https://user-images.githubusercontent.com/105614059/170884443-8fe8cf5e-b90f-48cc-9095-9b8cbe7c6780.png)

Reply:
just to be sure, test with the additional optional parameters? if they still work, then looks good to me

Reply:
Extra testing using quantities:
![image](https://user-images.githubusercontent.com/105614059/170884929-d159f636-4e6e-45f1-bd59-94a61c65b05e.png)

Reply:
for non stackable items that gets petty much ignored. check the augment fields. if you received an augmented item while using the wildcard, I think that's enough.

Reply:
I did not confirm the augments and values, but rather that the script still receives the call values.
![image](https://user-images.githubusercontent.com/105614059/170885163-746e1f3b-f7fc-4d4e-bbdb-f4a34fdc3c6c.png)

Reply:
from the picture I'd say it's doing fine. Thank you :+1: